src/network.ts

```diff
@@ -0,0 +1,22 @@
+/**
+ * what to do if fetch() generated an error
+ * @param response returned from the fetch() call
+ */
+const dealWithFetchError = (response: Response) => {
+  console.error(response);
+  throw new Error(JSON.stringify(response));
+};
+
+/**
+ * get data from a given url using fetch()
+ * @param url url for the webservice
+ */
+export const getWithFetch = async (url: string) => {
+  const response = await fetch(url, {
```

Comment:
how do we deal with nodejs environements? alotugh mostly used in browser one, we hould retain possibility to use it node.

Reply:
https://github.com/lquixada/cross-fetch maybe?

Reply:
If the same code doesn't need to run on both node and browser, I would use https://github.com/node-fetch/node-fetch on Node and the native fetch on browser.

Reply:
cross-fetch seems a bit more maintained than isomorphic-fetch, with some TS

Reply:
Spent the entire afternoon trying to make it work with cross-fetch.
It was breaking the tests, and somehow messing with nigiri.
When I tried with isomorphic-fetch, it was on the first try.
But I will take a new look at it today.

Reply:
So, it seems cross-fetch was working after all, but because nigiri was messed up, it was not passing the tests and I thought the problem was with cross-fetch. Today, I did a `nigiri stop --delete ; nigiri start --liquid` before running the tests and it worked on the first try. 🤦‍♂️

Reply:
Yes restarting Nigiri is often necessary :)